dll/win32/kernel32/winnls/string/nls.c

```diff
@@ -36,6 +37,10 @@ static const char UTF8Length[128] =
 /* First byte mask depending on UTF-8 sequence length. */
 static const unsigned char UTF8Mask[6] = {0x7f, 0x1f, 0x0f, 0x07, 0x03, 0x01};
 
+/* UTF-8 length to lower bound */
+static const unsigned long UTF8LBound[] =
+    {0, 0x80, 0x800, 0x10000, 0x200000, 0x2000000, 0xFFFFFFFF};
```

Comment:
Do we have tests covering these 5- and 6-byte encodings? (cause modern UTF-8 doesn't use them, so if we don't have to implement them, that'd be great.)

Reply:
Not have yet.

Reply:
Added 5- and 6-byte UTF-8 sequence tests: [89af984](https://github.com/reactos/reactos/commit/89af984b167aa9ed31b9a90abcd9c57920ba59e7)

Reply:
Thanks! However I only see "invalid" sequences in your new tests. What about "valid" ones?
My point being, standard UTF-8 does not have sequences longer than 4 bytes at all, so I would start out by assuming that conversion functions do not support them, unless a test shows otherwise.

Reply:
5-byte or more sequences became invalid by RFC 3629.

Reply:
Exactly. That's why I'm wondering whether we know for a fact that MultiByteToWideChar supports them at all.